### PR TITLE
Rl 131 - Creating an Interface for Create Relationships in Legacy system

### PR DESCRIPTION
**Pull request by zemlyanoi**
### JIRA Issue Link
https://osi-cwds.atlassian.net/browse/RL-131

### Technical Description
Creating an Interface for Create Relationships in Legacy system

### Tests
- [ ] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Configuration changes

N/A

### Technical debt

N/A

### CODE REVIEW

File: legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/dao/ClientDao.java [hunk 1]
```diff
@@ -31,6 +33,15 @@ public ClientDao(@CmsSessionFactory SessionFactory sessionFactory) {
     super(sessionFactory);
   }
 
+  @Override
+  public List<Client> findAll() {
```
Comment (sorych) [inline legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/dao/ClientDao.java]: is this really needed?

Comment (zemlyanoi) [reply legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/dao/ClientDao.java]: Fixed

File: legacy-data-access-services/src/main/java/gov/ca/cwds/cms/data/access/service/impl/clientrelationship/ClientRelationshipCoreService.java [hunk 2]
```diff
@@ -82,12 +88,62 @@ public ClientRelationship update(ClientRelationshipAwareDTO entityAwareDto)
...
     return searchClientRelationshipService.findRelationshipsByPrimaryClientId(clientId);
   }
+
+  @Override
+  public void createRelationship(ClientRelationshipDTO clientRelationshipDto)
+      throws DataAccessServicesException {
+    create(form(clientRelationshipDto));
+  }
+
+  private ClientRelationshipAwareDTO form(ClientRelationshipDTO clientRelationshipDto) {
```
Comment (sorych) [inline legacy-data-access-services/src/main/java/gov/ca/cwds/cms/data/access/service/impl/clientrelationship/ClientRelationshipCoreService.java]: I'd like to rename this method (buildAwareObject, compose, etc )

File: legacy-data-access-services/src/test/java/gov/ca/cwds/cms/data/access/service/impl/clientrelationship/ClientRelationshipCoreServiceTest.java [hunk 3]
```diff
@@ -30,6 +30,9 @@
 @PrepareForTest(PrincipalUtils.class)
 public class ClientRelationshipCoreServiceTest extends BaseUnitTest {
 
+  @Test
+  public void createRelationship() throws Exception {}
```
Comment (sorych) [inline legacy-data-access-services/src/test/java/gov/ca/cwds/cms/data/access/service/impl/clientrelationship/ClientRelationshipCoreServiceTest.java]: what is this?

File: legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java [hunk 4]
```diff
@@ -60,6 +60,7 @@
   name = "Client.findByFacilityIdAndChildId",
   query = Client.CHILDREN_BY_FACILITY_ID_BASE_QUERY + " AND c.identifier = :childId"
 )
+@NamedQuery(name = Client.NQ_ALL, query = "FROM gov.ca.cwds.data.legacy.cms.entity.Client")
```
Comment (sorych) [inline legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java]: is this used?

File: legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java [hunk 5]
```diff
@@ -68,6 +69,8 @@
 
   private static final long serialVersionUID = 783532074047017463L;
 
+  public static final String NQ_ALL = "Client.all";
```
Comment (sorych) [inline legacy-data-access/src/main/java/gov/ca/cwds/data/legacy/cms/entity/Client.java]: is this used?

**Pull request merged**